diff --git a/closed/DDR.gmk b/closed/DDR.gmk
index 77a9d9b136..72a09bd5cd 100644
--- a/closed/DDR.gmk
+++ b/closed/DDR.gmk
@@ -70,6 +70,9 @@ DDR_STRUCTURES_MARKER := $(DDR_SUPPORT_DIR)/gensrc-structures.done
 # Build the tools we will need.
 $(eval $(call SetupJavaCompilation,BUILD_DDR_TOOLS, \
 	SETUP := GENERATE_OLDBYTECODE, \
+	ADD_JAVAC_FLAGS := \
+		--add-exports java.base/jdk.internal.org.objectweb.asm=ALL-UNNAMED \
+		--add-exports java.base/jdk.internal.org.objectweb.asm=jdk.compiler.interim, \
 	BIN := $(DDR_TOOLS_BIN), \
 	CLASSPATH := $(JDK_OUTPUTDIR)/modules/java.base, \
 	SRC := $(DDR_VM_SRC_ROOT), \
